staging/src/k8s.io/client-go/pkg/apis/clientauthentication/types.go

```diff
@@ -75,3 +80,42 @@ type Response struct {
 	// Code is the HTTP status code returned by the server.
 	Code int32
 }
+
+// Cluster contains information to allow an exec plugin to communicate
+// with the kubernetes cluster being authenticated to.
```

Comment:
I missed some of the discussion around this, so asked a question at https://github.com/kubernetes/enhancements/pull/1711/files#r445956764, echoing here:

I can see use cases for cluster hostname and custom config bits, but I was surprised by the ServerName and CAData bits... are we expecting exec plugins to call out to the server we are asking them to provide credentials for?

If so, don't we need to include ProxyURL and InsecureSkipTLSVerify?

Reply:
is there a reason not to just use the serialized version of the cluster stanza from the kubeconfig file more or less as-is?

Reply:
> I was surprised by the ServerName and CAData bits... are we expecting exec plugins to call out to the server we are asking them to provide credentials for?

Yes, similar to what `oc login` does today to determine the location of the OAuth server (i.e. read this well known config map that tells you something dynamic about the cluster).

> If so, don't we need to include ProxyURL and InsecureSkipTLSVerify?

The per-cluster `ProxyURL` was added in v1.19 without me noticing, but yeah, we should include that.  I left out `InsecureSkipTLSVerify` on purpose to discourage its use...

> is there a reason not to just use the serialized version of the cluster stanza from the kubeconfig file more or less as-is?

I had that in the original draft of the KEP updates.  See https://github.com/kubernetes/enhancements/pull/1711#discussion_r417451744.  @mikedanese leaned towards a new struct, and I liked the idea of just having a single well-known extension key to pass through instead of just dumping everything.

Reply:
Opened a new PR into the KEP to try to address this design update: https://github.com/kubernetes/enhancements/pull/1927.